pkg/apis/core/types.go

```diff
@@ -2102,6 +2112,11 @@ type Lifecycle struct {
 	// or until the termination grace period is reached.
 	// +optional
 	PreStop *Handler
+	// Type describes the startup/shutdown behaviour of the container.
+	// One of Standard, Sidecar.
+	// Defaults to Standard.
+	// +optional
+	Type *LifecycleType
```

Comment:
do we need to add lifecycle info to container status as well? this would provide more information to evaluate pod status.